### ISSUE TITLE
Change `UiId` to `WidgetId` to distinguish more clearly from `CanvasId`.

### ISSUE THREAD

**Issue by mitchmindtree** [PistonDevelopers/conrod#430]
Seeing as both widgets and canvasses have their own caches, perhaps it would be better to more clearly distinguish the `UiId` (which is an index into the `widget_cache`) from the `CanvasId` (an index into the `canvas_cache`) and call it `WidgetId` instead.

**Comment by mitchmindtree** [comment]
We'll also need an enum to distinguish between the IDs for checking user input capturing (i.e. a Canvas may capture mouse input when being dragged, or key input for shortcuts). Perhaps we can use UiId for this:
```Rust
pub enum UiId {
    Widget(WidgetId),
    Canvas(CanvasId),
}
```

**Comment by themooserooster** [comment]
I'd like to give this one a crack.

**Comment by mitchmindtree** [comment]
@themooserooster I've already started on this - it's turning out to involve a lot of other small changes with the capturing of different Ui elements etc, I'll probably just include it in my next PR along with the Floating canvas. Thanks for the intent though!

**Comment by themooserooster** [comment]
Ok, fair enough. I'm just trying to get started with Rust, so this seemed like a project worth contributing to.

**Comment by mitchmindtree** [comment]
Closed by #433
